Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -15967,7 +15967,7 @@ void SpeculativeJIT::compileToPropertyKeyOrNumber(Node* node)
 
     addSlowPathGenerator(slowPathCall(slowCases, this, operationToPropertyKeyOrNumber, resultRegs, LinkableConstant::globalObject(*this, node), argumentRegs));
 
-    jsValueResult(resultRegs, node, DataFormatJSCell, UseChildrenCalledExplicitly);
+    jsValueResult(resultRegs, node, DataFormatJS, UseChildrenCalledExplicitly);
```

Comment:
Since `ToPropertyKeyOrNumber` has more speculated types `SpecString | SpecSymbol | SpecBytecodeNumber `, why is it OK to downgrade from `DataFormatJSCell` to `DataFormatJS `?
```
DataFormatJSCell = DataFormatJS | DataFormatCell,
```

Reply:
`DataFormatJSCell` means we know the value is boxed as a `JSCell`, which only has meaning on 32-bit.